### PR TITLE
Only set session token variable (AWS_SESSION_TOKEN) if it was present in the session

### PR DESCRIPTION
**Pull request by resios**
This prevents situations where the session token variable is set but it contains invalid data.
This can lead to errors in the default [environment credential provider](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/auth/EnvironmentVariableCredentialsProvider.java#L46) used by the AWS Java SDK.

### CODE REVIEW

File: runtime.go [hunk 1]
```diff
@@ -394,9 +392,14 @@ func getSessionOrDefaultCreds() map[string]string {
 
 			result["key"] = creds.AccessKeyID
 			result["secret"] = creds.SecretAccessKey
-			result["sessiontoken"] = creds.SessionToken
+			if creds.SessionToken != "" {
+				result["sessiontoken"] = creds.SessionToken
+			} else {
+				delete(result, "sessiontoken")
```
Comment (sanathkr) [inline runtime.go]: You have already removed `sessiontoken` from the initialization. Why is this needed?

Comment (resios) [reply runtime.go]: Good catch

**Comment by sanathkr** [comment]
Appreciate doing this!
